test/libsyncengine/propagation/executor/executorworker/testexecutorworker.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "testexecutorworker.h"
+
+using namespace CppUnit;
+
+namespace KDC {
+
+static const SyncTime defaultTime = 1654788079;
+static const int64_t defaultSize = 1654788079;
+
+void TestExecutorWorker::setUp() {
+    // Create SyncPal
+    bool alreadyExists = false;
+    std::filesystem::path parmsDbPath = Db::makeDbName(alreadyExists, true);
+    ParmsDb::instance(parmsDbPath, "3.4.0", true, true);
+
+    SyncPath syncDbPath = Db::makeDbName(1, 1, 1, 1, alreadyExists);
+    std::filesystem::remove(syncDbPath);
+    _syncPal = std::make_shared<SyncPal>(syncDbPath, "3.4.0", true);
+    _testObj = std::make_shared<ExecutorWorker>(_syncPal, "name", "short name");
+
+    _syncPal->_syncDb->setAutoDelete(true);
+
+    /**
+     * Initial FS state:
+     *
+     *            root
+     *             |
+     *             A
+     *        _____|______
+     *       |           |
+     *      AA          AB
+     *      |
+     *     AAA
+     */
+
+    // Setup DB
+    DbNodeId dbNodeIdA;
+    DbNodeId dbNodeIdAA;
+    DbNodeId dbNodeIdAB;
+    DbNodeId dbNodeIdAAA;
+
+    bool constraintError = false;
+    DbNode dbNodeA(0, _syncPal->_syncDb->rootNode().nodeId(), Str("A"), Str("A"), "lA", "rA", defaultTime, defaultTime,
+                   defaultTime, NodeType::NodeTypeDirectory, 0, std::nullopt);
+    _syncPal->_syncDb->insertNode(dbNodeA, dbNodeIdA, constraintError);
```

Comment:
```suggestion
    _syncPal->syncDb()->insertNode(dbNodeA, dbNodeIdA, constraintError);
```
You should only use the public interface, whenever possible.